User story 30 pd

philjdelong opened the pull request:
## Description
Resolves #32 

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Refactor (non-breaking change which maintains existing functionality)
  (order_controller, method: .update)
- [x] New feature (no-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## RSpec Results
**ALL UNCOVERED CODE IS EITHER:
     - SAD PATH
     - Part 2 - STORIES**
```
Finished in 5.75 seconds (files took 2.36 seconds to load)
161 examples, 0 failures

Coverage report generated for RSpec to /Users/phildelong/turing/mod2/projects/monster_shop_part_1/coverage. 2
029 / 2036 LOC (99.66%) covered.
```

philjdelong committed:
Pass story 30 and write method for unfulled upon order cancellation

philjdelong committed:
Pass story 30 and write method for unfulled upon order cancellation

philjdelong committed:
Merge branch 'unser_story_30_pd' of github.com:philjdelong/monster_shop_part_1 into unser_story_30_pd

philjdelong committed:
Include conditional for user status within update method

philjdelong committed:
Pass all tests for 30. All uncovered lines will be covered later

mel-rob reviewed (approved):
Looks great, thanks!

DanielEFrampton commented on app/controllers/orders_controller.rb:
```diff
@@ -36,10 +36,20 @@ def create
     order.update(update_params)
-    unless order.save
-      flash[:error] = order.errors.full_messages.to_sentence
+    if order.save
+      if order.cancelled?
+        flash[:notice] = "Your Order has been Cancelled"
+        redirect_to "/profile"
+      elsif current_user.admin?
+        redirect_to "/admin"
+      elsif current_user.merchant?
+        redirect_to "/merchant"
+      elsif current_user.default?
+        redirect_to "/profile"
+      else
+        flash[:error] = order.errors.full_messages.to_sentence
+      end
```
That works! Let's refactor it later in the project to pull some of this stuff into helper methods.

DanielEFrampton committed:
Update show.html.erb

Removed query string from cancel order link that didn't end up getting utilized.

DanielEFrampton commented on app/models/item_order.rb:
```diff
@@ -15,4 +17,10 @@ def change_order_to_packaged
   end
+
+  def change_to_unfulfilled
+    if self.fulfilled?
+      item.update(inventory: (item.inventory + quantity))
+    end
```
I'm a little confused about how the order of these things works, but the test looks solid so I guess it's working! (and I was there when we designed this, ha!)

DanielEFrampton reviewed (approved):
Thanks for tackling this one, it was really confusing. Solid test.

DanielEFrampton committed:
Merge branch 'master' into unser_story_30_pd
